tap/extensions/http/handlers.go

```diff
@@ -21,9 +21,30 @@ func filterAndEmit(item *api.OutputChannelItem, emitter api.Emitter, options *ap
 		FilterSensitiveData(item, options)
 	}
 
+	replaceForwardedFor(item)
+
 	emitter.Emit(item)
 }
 
+func replaceForwardedFor(item *api.OutputChannelItem) {
+	if item.Protocol.Name != "http" {
+		return
+	}
+
+	request := item.Pair.Request.Payload.(api.HTTPPayload).Data.(*http.Request)
+
+	forwardedFor := request.Header.Get("X-Forwarded-For")
+	if forwardedFor == "" {
+		return
+	}
+
+	ips := strings.Split(forwardedFor, ",")
+	lastIP := strings.TrimSpace(ips[0])
+
+	item.ConnectionInfo.ClientIP = lastIP
+	item.ConnectionInfo.ClientPort = ""
```

Comment:
```suggestion
        // the client port we see for forwarded traffic is not the original port
	item.ConnectionInfo.ClientPort = ""
```

Reply:
theres probably a better way to word this than what I came up with

Reply:
Thanks, I will adopt.